src/lib/modules/console/suggestions.ts

```diff
@@ -1,32 +1,69 @@
 let utils = require('../../utils/utils');
 
-class Suggestions {
-  constructor(embark, options) {
+type Events = {
+  on: any,
+  request: any
+}
+
+type EmbarkApi = {
+  events: Events,
+  registerAPICall: any
+}
```

Comment:
This should be a type provided by `@embark/core` as we have many other places that take an `Embark API`.

OTOH, I was actually thinking about whether we want to rethink the Embark API construct and instead introduced dedicated services for certain thing (EventService, etc) and then introduce an injection system so that we can do something like:

```
export class Suggestions {
  constructor(private eventBus: Eventbus, private configService: ConfigService) {}
}
```

Also notice that in TypeScript, the above is a shorthand for:

```
export class Suggestions {
   private eventBus: EventBus;
   private configService: ConfigService;

  constructor(eventBus: Eventbus, configService: ConfigService) {
    this.eventBus = eventBus;
    this.configService = configService;
  }
}
```
Which could save us a lot of typing as well.

Anyways, for now I think we should put `EmbarkAPI` somewhere in side `core` and then import it from there.

Reply:
Those type will probably end up being classes quite soon (once the core has been done)
In the mean type should we add a folder types where we put them.
As you said they are transient